### PR TITLE
[dependabot][mergify] automate merge and add reviewers

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

- Add reviewers
- Automerge if CI is happy. Although, if no approval is needed why do we need the reviewers?

## Why is it important?

Look mum no hands!

## Issue

See https://github.com/elastic/integrations/pull/1291#issuecomment-876963997

Examples with mergify -> https://docs.mergify.io/examples/#example-rules

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -6,6 +6,8 @@ updates:
     schedule:
       interval: "daily"
     labels:
-      - dependencies
-      - go
+      - automation
+    reviewers:
+      - "mtojek"
+      - "jsoriano"
```
Comment (jsoriano) [inline .github/dependabot.yml:12]: I'd prefer to put a [github] team here so we can more easily manage notifications and we don't need need to change this if the people related to the project changes. But we can go on like this by now.

**Review by jsoriano — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (v1v) [reply .github/dependabot.yml:12]: Sure thing.

BTW, we are running a PoC with a `oncall` github team, so the person oncall need to subscribe to the team and remove other team members. The idea is to reduce the overhead/notifications for other team members when they are not on-call, but we are still at the very beginning of this evaluation :)

Comment (jsoriano) [reply .github/dependabot.yml:12]: Oh, this sounds great to ping with the person of call for issues related to builds/CI. 

> the person oncall need to subscribe to the team and remove other team members.

Would it be possible to do it automatically by checking the Pagerduty calendars?

**Review by kuisathaverat — APPROVED** [review]

**Review by mtojek — APPROVED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (v1v) [reply .github/dependabot.yml:12]: I just answered you in the initial discussion we have opened in the obs-robots team regarding this process :)

Comment (jsoriano) [reply .github/dependabot.yml:12]: Thanks!

File: .github/dependabot.yml [hunk 2]
```diff
@@ -6,6 +6,8 @@ updates:
     schedule:
       interval: "daily"
     labels:
-      - dependencies
-      - go
+      - automation
+    reviewers:
+      - "mtojek"
```
Comment (v1v) [inline .github/dependabot.yml:11]:
```suggestion
      - "elastic/integrations-developer-experience"
```

File: .github/dependabot.yml (hunk 1 above)
Comment (v1v) [reply .github/dependabot.yml:12]:
```suggestion
```

**Review by jsoriano — APPROVED** [review]

**Review by mtojek — APPROVED** [review]

**Pull request merged**